src/data/finsupp/lex.lean

```diff
@@ -0,0 +1,68 @@
+/-
+Copyright (c) 2022 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.pi.lex
+import data.finsupp.basic
+
+/-!
+# Lexicographic order on finitely supported functions
+
+This file defines the lexicographic order on `finsupp`.
+-/
+
+namespace finsupp
+
+variables {α N : Type*} [has_zero N]
+
+open_locale classical
```

Comment:
Does removing this prevent things being `noncomputable` below?

Reply:
Removing `open_locale classical` prevents union in finsupp from existing.

Reply:
Then you're missing `decidable_eq α` on a lemma somewhere. If you find yourself addnig `open_locale classical` to make a _lemma statement_ compile, you've made a wrong turn.

Reply:
Should I then also add `decidable_pred (λ (a : α), ⇑f a ≠ ⇑g a)`, which now Lean asks?

Reply:
(Or maybe `decidable_eq N`?)

Reply:
@eric-wieser, I take it that you are happy with the `decidable_eq` assumption, instead of the `ne` version that is needed?  Is there some rationale between preferring one over the other (besides simplicity of stating)?